gax/src/gax.ts

```diff
@@ -67,27 +70,70 @@ import {toLowerCamelCase} from './util';
 
 /**
  * Per-call configurable settings for retrying upon transient failure.
+ * @implements {RetryOptionsType}
  * @typedef {Object} RetryOptions
- * @property {String[]} retryCodes
+ * @property {number[] | (function)} retryCodesOrShouldRetryFn
  * @property {BackoffSettings} backoffSettings
+ * @property {(function)} getResumptionRequestFn
  */
 export class RetryOptions {
-  retryCodes: number[];
+  retryCodesOrShouldRetryFn: number[] | ((error: GoogleError) => boolean);
   backoffSettings: BackoffSettings;
-  constructor(retryCodes: number[], backoffSettings: BackoffSettings) {
-    this.retryCodes = retryCodes;
+  getResumptionRequestFn?: (request: RequestType) => RequestType;
+  constructor(
+    retryCodesOrShouldRetryFn: number[] | ((error: GoogleError) => boolean),
+    backoffSettings: BackoffSettings,
+    getResumptionRequestFn?: (request: RequestType) => RequestType
+  ) {
+    this.retryCodesOrShouldRetryFn = retryCodesOrShouldRetryFn;
     this.backoffSettings = backoffSettings;
+    this.getResumptionRequestFn = getResumptionRequestFn;
+  }
+}
+
+/**
+ * Helper function to reduce the type checking for this variable to one spot
+ * @param retryCodesOrShouldRetryFn
+ * @returns
+ */
+export function isRetryCodes(
+  retryCodesOrShouldRetryFn: number[] | ((error: GoogleError) => boolean)
+) {
+  if (Array.isArray(retryCodesOrShouldRetryFn)) {
+    return true;
+  } else if (retryCodesOrShouldRetryFn instanceof Function) {
```

Comment:
nit(style): no need to have `else` after `return`

Reply:
obsolete